### DISCUSSION TITLE
Suggestions and Issues Regarding the VocaDB Player (Non-Beta)

### DISCUSSION THREAD

**Discussion by CatgirlFrostmoon** [VocaDB/community#171] [Ideas]
This is something of a continuation of riipah's [previous thread](https://github.com/VocaDB/vocadb/issues/1114), as you can probably tell from the title. ;3 Now that it's out of beta, it's a bit easier to see how the player actually functions during common VDB usage, and while it's definitely functional, it's a bit strange.

1. When I refresh, the menu pops back despite having been closed prior. It also appears in a new tab, despite the original tab having the menu toggled off. Could we possibly make the on/off toggle more permanent? I'd like it to stay closed personally, instead of reappearing during refreshes or through new tabs.

2. Related to the above, it might make sense to have a temporary close button on the hamburger menu on the bar, and a permanent close button in the "My Account" menu (where the close button currently is). Admittedly, I'm not sure where the first reopen button would be.

3. Closing the bottom bar doesn't remove the playing/previous video thumbnail at the bottom right of the screen, while it really should. The videos look fine with the bar, but they're quite awkward without it.

4. Relatedly, you can't close the bottom right video without the bar open. 

5. Also, switching between Youtube/Soundcloud/etc versions of the same video opens the bottom right video; I'm not sure if this is a good idea? Might want to make this a toggle.

6. It seems unable to play age-restricted Youtube songs (tested on the original ME!ME!ME!). I'm not sure if this is fixable on our end.

7. The video image (seen in screenshot) is...weird, to say the least. I have it set so that Youtube's my preferred platform here, and while that wasn't an issue before, it's certainly causing some oddities here. First, as the screenshot shows, the image/video defaults to Youtube _even when the Youtube link is unavailable_. Second, when I click on a different version (such as Soundcloud), the image/embed doesn't actually change until I click on the little play button. Notably, everything else (Soundcloud -> NicoNico, for instance) swaps videos/images immediately, but they all seem to default to the Youtube embed? It's strange.
![weh](https://user-images.githubusercontent.com/98185854/187780449-f5edf077-8790-4ea9-925f-204bf7b24c21.JPG)

**Comment by ycanardeau** [comment]
I just don't have motivation to implement most of these features, because I'd like to concentrate on implementing more useful and/or important features. I already spent much more time on thinking about UI/UX than you did (making several experiments, reading books and blog posts, using multiple media players and websites and etc.), and I'm quite happy and satisfied with the current implementation.

> 1.

Actually, the "Show/Hide bottom bar" feature is not intended to be used extensively. I'm quite sure you'll be accustomed to the new bottom bar before the permanent option being provided. This can be compared to the fact that the top and side bars can't be closed in the first place. I'd rather suggest you buy a 56-pixel-taller (which is the height of the bottom bar) display than you have, which would solve your space issue elegantly.

> 4.

For future reference, [it's *not* allowed to display overlays, frames, or other visual elements in front of any part of a YouTube embedded player](https://developers.google.com/youtube/terms/required-minimum-functionality#overlays-and-frames), **including player controls**. Try hard refresh by pressing Ctrl+F5 if you get an "Service Unavailable" error when loading the page.

> 6.

Can you please open an issue on [VocaDB/nostalgic-diva](https://github.com/VocaDB/nostalgic-diva/issues) instead?

> 7.

We must "lazy-load" videos to prevent cookies from being set unless an embedded video is played. Search the web for "cookie consent banner", if you're curious why we need to do like this (Spoiler: this was one of the main motivations to implement the bottom bar and the mini player). I'm not sure if there is a way to obtain a video preview without loading a video. The implementation of [react-player](https://github.com/cookpete/react-player) would help as a reference.

Of course, as always, pull requests are welcome if someone else wants to implement these features.

Quote from [riipah's blog post](https://web.archive.org/web/20210724203408/https://blog.vocadb.net/about-feature-requests/):

> VocaDB will never satisfy every possible need. It’s better to keep some focus rather than try to please everyone. I simply don’t have the resources to maintain a huge project alone. I’m actually encouraging other developers to build related services around VocaDB – websites that use VocaDB as a data source through its APIs.

Read each blog post 39 times.

**Comment by CatgirlFrostmoon** [comment]
> Actually, the "Show/Hide bottom bar" feature is not intended to be used extensively. I'm quite sure you'll be accustomed to the new bottom bar before the permanent option being provided. This can be compared to the fact that the top and side bars can't be closed in the first place. I'd rather suggest you buy a 56-pixel-taller (which is the height of the bottom bar) display than you have, which would solve your space issue elegantly.

The issue with the bottom bar isn't space, it's that it's a highly-attention grabbing UI element with a strong contrast to everything else. It's distracting, and I can't focus with it around. It would be very helpful if we could turn it persistently off for site accessibility. I know I'm not the only person with issues here, either -- a person on the Discord mentioned that they had wanted to start a similar thread to mine, but are having difficulties posting here. As such, while you may have "spent more time thinking about UI" than I ever have, because of my specific disabilities, I tend to notice certain accessibility flaws before other people. By working together, we can make a better product.

> For future reference, [it's _not_ allowed to display overlays, frames, or other visual elements in front of any part of a YouTube embedded player](https://developers.google.com/youtube/terms/required-minimum-functionality#overlays-and-frames), **including player controls**. Try hard refresh by pressing Ctrl+F5 if you get an "Service Unavailable" error when loading the page.

Ah, Youtube. It might be best to not have the bottom thumbnail appear at all if the bar is closed, then, or perhaps a button _to the side_ of the thumbnail (not on it, beside it) could work.

> Can you please open an issue on [VocaDB/nostalgic-diva](https://github.com/VocaDB/nostalgic-diva/issues) instead?

I'll do that in a bit. EDIT: Done.

> We must "lazy-load" videos to prevent cookies from being set unless an embedded video is played. Search the web for "cookie consent banner", if you're curious why we need to do like this (Spoiler: this was one of the main motivations to implement the bottom bar and the mini player). I'm not sure if there is a way to obtain a video preview without loading a video. The implementation of [react-player](https://github.com/cookpete/react-player) would help as a reference.

That's fair. I quite appreciate the explanation, so thank you.


EDIT: Removed the numbers and added in quotes. I wasn't expecting Github to automatically convert my 1/4/6/7 to 1/2/3/4.

**Comment by ycanardeau** [comment]
First of all, I'm not against making it possible to turn the bottom bar persistently off, and I understand that there are some people who tend to notice certain accessibility flaws before other people. I just don't have resources to implement it alone, because we have more important and urgent issues to resolve first. It only takes one hour or two to make a suggestion, but it takes 10-20 hours, or even more, (unpaid) to implement it. As I said, "[pull requests](https://docs.github.com/en/pull-requests) are welcome". I'll lock the conversation for now.

P.S. Thanks for opening an issue at VocaDB/nostalgic-diva#2.
